library/src/main/scala/scala/scalajs/LinkingInfo.scala

```diff
@@ -108,4 +108,35 @@ object LinkingInfo {
   def assumingES6: Boolean =
     linkingInfo.assumingES6
 
+  /** The value of the global JavaScript `this`.
+   *
+   *  This returns the value that would be obtained by writing `this` at the
+   *  top-level of the JavaScript file generated by Scala.js. In most
+   *  JavaScript environments, this is equivalent to the *global object*, but
+   *  it is not necessarily the case. For example, on Node.js, `this` is the
+   *  object representing the `exports` of the current module.
+   *
+   *  Using this value should be rare, and mostly limited to writing code
+   *  detecting what the global object is. For example, a typical detection
+   *  code looks like:
+   *  {{{
+   *  val globalObject = {
+   *    import js.Dynamic.{global => g}
+   *    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object)) {
+   *      // Node.js environment detected
+   *      g.global
+   *    } else {
+   *      // In all other well-known environment, we can use the global `this`
+   *      scala.scalajs.LinkingInfo.globalThis
+   *    }
+   *  }
+   *  }}}
+   *  Note that the above code is not comprehensive, as there can be JavaScript
+   *  environments where the global object cannot be fetched neither through
+   *  `global` nor `this`. If your code needs to run in such an environment, it
+   *  is up to you to use an appropriate detection procedure.
+   */
+  @inline
+  def globalThis: js.Dynamic = linkingInfo.globalThis
+
```

Comment:
Yes, I guess we should investigate that. Given the set we see now, perhaps `js.meta._`, since these things are a bit, well, meta. `new.target` is even called a meta-property in the ES spec. I'm afraid `js.meta._` would be confused with `scala.meta`, though, so strategically speaking it's probably not a good idea.

`new.target` I'm afraid will need *both* a primitive (to be called from the constructor of JS classes) and a set of `js.NewTargetFunctionN` (to be called from JS `function`s). The former is the most "critical", though. Using it in `function`s should be pretty niche even compared to using it in classes, which itself should be pretty rare. So we're not a time pressure to come up with a good design for that one.